src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c

```diff
@@ -1307,8 +1307,14 @@ static ssize_t
 uuid_show(struct device *dev, struct device_attribute *attr, char *buf)
 {
 	struct xocl_hwmon_sdm *sdm = dev_get_drvdata(dev);
+	int count = 0, i = 0;
 
-	return sprintf(buf, "0x%llx\n", sdm->bdinfo.uuid);
+	count += sprintf(buf, "0x");
+	for (i = 0; i < strlen(sdm->bdinfo.uuid); i++)
```

Comment:
Using strlen() does not look right. What if there is 0 in uuid[] ?

Reply:
It makes also the loop complexity O(n2) because strlen is already O(n)...

Reply:
@rajkumar-xilinx I assume big endian in bdinfo.uuid. you can use %pUb (lower case) or %pUB(upper case). The output buffer format is "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee" and the length should be UUID_STRING_LEN + 1.

Reply:
Thanks @houlz0507 your suggestion worked well.